### DISCUSSION TITLE
[Help]: Is there any way to prevent scrolling in a certain direction?

### DISCUSSION THREAD

**Discussion by clementoriol** [davidjerleke/embla-carousel#855] [Help] [answered]
### Summary

Hello, I'm looking for a way to programatically prevent scrolling. Here's what I need : 
- When the user starts scrolling, trigger an event or a callback
- Get some scroll data in the callback, like the scroll direction
- Let me run arbitrary business logic (is the user allowed to scroll in this direction ?)
- Prevent the scroll or allow it 

Is there some way to implement this already or should it be a feature request ?

I've tried : 
- the 'scroll' event, but I can't seem to prevent the scroll in a clean way.
- the `watchDrag` callback. I can prevent the scroll but I can't find how to calculate the scroll direction from the callback args.

Any ideas ?

Thank you

### If applicable, which variants of Embla Carousel are relevant to this question?

- [X] embla-carousel **(Core)**
- [X] embla-carousel-react
- [ ] embla-carousel-vue
- [ ] embla-carousel-svelte
- [ ] embla-carousel-autoplay
- [ ] embla-carousel-auto-scroll
- [ ] embla-carousel-solid
- [ ] embla-carousel-auto-height
- [ ] embla-carousel-class-names
- [ ] embla-carousel-docs **(Documentation)**
- [ ] embla-carousel-docs **(Generator)**

**Comment by davidjerleke** [comment, marked as answer]
Thanks for your question @clementoriol.

What you want to do isn't possible at the time of writing. However, there's an upcoming feature called event interceptors that will solve this in the future. Interceptors will be available for the `pointerMove` event. It's mentioned here:

- #841 

I believe this is a duplicate of: 

- #749

**Reply by clementoriol** [reply]
Thank you @davidjerleke for your input on this. Sorry I didn't found the similar issue, and glad this is on the roadmap ☀️
